Source/Core/Core/Core.cpp

```diff
@@ -146,7 +154,7 @@ bool IsRunningInCurrentThread()
 
 bool IsCPUThread()
 {
-	return (s_cpu_thread.joinable() ? (s_cpu_thread.get_id() == std::this_thread::get_id()) : !s_is_started);
+	return s_cpu_thread.joinable() ? tls_is_cpu_thread : !s_is_started;
```

Comment:
So should I call the DeclareAsCPUThread() there for a small amount of time, and then undeclare it? Or should we move that code into the CpuThread?